Sources/SwiftProcessing/Core/Sketch3D/TransitionSCNNode.swift

```diff
@@ -2,42 +2,51 @@ import UIKit
 import SceneKit
 
 class TransitionSCNNode: SCNNode {
-    
+
     var parentTransitionNodeTag: String = ""
     var tag: String = ""
     var availabletransitionNodes: [TransitionSCNNode] = []
     var currentShapes: [SCNNode] = []
     var availableShapeNodes: [String : [SCNNode]] = [:]
     var currentShapeNodes: [String : [SCNNode]] = [:]
-    
+
     func hasNoShapeNodes() -> Bool {
         return self.currentShapes.count == 0
     }
-    
+
     func hasAvailableTransitionNodes() -> Bool {
         return self.availabletransitionNodes.count > 0
     }
-    
+
     func getNextAvailableTransitionNode() -> TransitionSCNNode {
         return self.availabletransitionNodes.popLast()!
     }
-    
+
+    func deleteTransitionNode() {
+        for node in self.availabletransitionNodes {
```

Comment:
was this the missing piece to eliminate the duplicate shapes appearing / sketch to crash? 🥳